ir/src/trace_column.rs

```diff
@@ -0,0 +1,61 @@
+use super::TraceSegment;
+
+/// Describes a column in the execution trace by the trace segment to which it belongs and its
+/// index within that segment.
+#[derive(Debug, Copy, Clone)]
+pub struct TraceColumn {
+    trace_segment: TraceSegment,
+    col_idx: usize,
+}
+
+impl TraceColumn {
+    /// Creates a [TraceColumn] in the specified trace segment at the specified index.
+    pub fn new(trace_segment: TraceSegment, col_idx: usize) -> Self {
+        Self {
+            trace_segment,
+            col_idx,
+        }
+    }
+
+    /// Gets the trace segment of this [TraceColumn].
+    pub fn trace_segment(&self) -> TraceSegment {
+        self.trace_segment
+    }
+
+    /// Gets the column index of this [TraceColumn].
+    pub fn col_idx(&self) -> usize {
+        self.col_idx
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct TraceColumnGroup {
+    trace_segment: TraceSegment,
+    size: usize,
+    start_col_idx: usize,
```

Comment:
nit: instead of `start_col_idx` I'd prefer `offset`. It's more consistent with the way we've done things in the Miden VM code